client/store/cart.js

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable no-case-declarations */
+import axios from "axios";
+const TOKEN = "token";
+
+const SET_CART = "SET_CART";
+const ADD_TO_CART = "ADD_TO_CART";
+const REMOVE_FROM_CART = "REMOVE_FROM_CART";
+const REMOVE_ALL_FROM_CART = "REMOVE_FROM_CART";
+
+//ACTION CREATORS
+export const _setCart = (cart) => {
+  return {
+    type: SET_CART,
+    cart,
+  };
+};
+
+export const _addToCart = (product) => {
+  return {
+    type: ADD_TO_CART,
+    product,
+  };
+};
+
+export const _removeFromCart = (product) => {
+  return {
+    type: REMOVE_FROM_CART,
+    product,
+  };
+};
+
+export const _removeAllFromCart = (product) => {
+  return {
+    type: REMOVE_ALL_FROM_CART,
+    product,
+  };
+};
+
+//THUNK CREATORS
+export const setCart = () => {
+  return async (dispatch) => {
+    const token = window.localStorage.getItem(TOKEN);
+    const response = await axios.get(`/api/cart`, {
+      headers: {
+        authorization: token,
+      },
+    });
+    const cart = response.data;
+    dispatch(_setCart(cart.lineitems));
+    console.log("THUNK set cart", cart.lineitems);
+  };
+};
+
+export const addToCart = (product) => {
+  return async (dispatch) => {
+    const token = window.localStorage.getItem(TOKEN);
```

Comment:
Same thing here re: non-users. Something will have to change here for addToCart to work if you're not signed up/logged in